plugins/module_utils/mh/mixins/cmd.py

```diff
@@ -166,7 +173,12 @@ def run_command(self, extra_params=None, params=None, process_output=None, *args
             self.update_output(force_lang=self.force_lang)
             options['environ_update'] = env_update
         rc, out, err = self.module.run_command(self.vars.cmd_args, *args, **options)
-        self.update_output(rc=rc, stdout=out, stderr=err)
+        if publish_rc:
+            self.update_output(rc=rc, stdout=out, stderr=err)
+        if publish_out:
+            self.update_output(rc=rc, stdout=out, stderr=err)
+        if publish_err:
+            self.update_output(rc=rc, stdout=out, stderr=err)
```

Comment:
So what's the difference between `publish_rc`, `publish_out` and `publish_err`? I think you need to adjust the `update_output` calls a bit ;)

Reply:
Yeah, it was that late at night. Wrapping this up today.